### DISCUSSION TITLE
Code management for the coupled/uncoupled Prototype test suite

### DISCUSSION THREAD

**Discussion by DeniseWorthen** [ufs-community/ufs-weather-model#891] [Announcements]
The standalone and coupled "P" regression tests (RTs) contain the feature set of the current prototype. As of late Oct 2021, the current feature set is P7. Modifications of the existing P7 RTs can take two forms:

1. Improvements, bugfixes or modifications to existing features or code. Examples:

   - Fix to code base: Surface emissivity fixes (#821)
   - Improvement of P7 feature: scale aware CA (#888)
   - Modification for P7 feature: NoahMP additional fields (FV3 PR [# 387](https://github.com/NOAA-EMC/fv3atm/pull/387))

   In Case 1), there are no new features being turned on. These code modifications can proceed using the existing P7 test, ensuring that the updated feature set can still pass the full RT suite, which includes the operational requirements (debug, threading etc). 

2. Turning on existing features or add a new feature in the code base. Examples:

   - Turn on FLAKE (no PR, no coupled SDF)
   - Turn on CICE albedos (namelist change only)
   - Turn on GOCART (no current RT for coupled model, limited to Hera only).

   In Case 2), there are new features being turned on or added. Adding these features in the P7 suite will implicitly update the P7 suite to the base P8 suite. Note that at onset, the P8 suite may not contain the final P8 feature set. It will be the P7 feature set "+". However, the tests will be renamed P8 since they represent a new feature set targeted to the next Prototype.

3. Managing the Commit Order for Cases 1) and 2)

- Case 1): PRs which improve or modify the existing P7 test suite can be committed when ready.

- Case 2): A PR which implements an initial feature(s) for P8 needs to be prepared. The following steps are pre-cursors to that PR: 

     - If a new SDF is required, this must be provided by the feature developer as well as a list of the required namelist settings.
     - The oRT needs to be run with these features turned on. 
     - Once all oRTs pass with the initial P8 feature(s), the PR to update the P7 test suite to P8 can be made. Previous P7 tests will not be updated once the initial P8 (ie, P7+new feature) tests pass the oRTs and are implemented in the full RT test.

**Comment by SMoorthi-emc** [comment]
FYI,
All of my own coupled tests have cice albedo turned on and do reproduce on restart and do run fine in debug mode.
